mailgun/resource_mailgun_credential.go

```diff
@@ -0,0 +1,173 @@
+package mailgun
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mailgun/mailgun-go/v4"
+)
+
+func resourceMailgunCredential() *schema.Resource {
+	log.Printf("[DEBUG] resourceMailgunCredential()")
+
+	return &schema.Resource{
+		CreateContext: resourceMailgunCredentialCreate,
+		Read:          resourceMailgunCredentialRead,
+		Update:        resourceMailgunCredentialUpdate,
+		Delete:        resourceMailgunCredentialDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceMailgunCredentialImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"login": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"password": {
+				Type:      schema.TypeString,
+				ForceNew:  false,
+				Required:  true,
+				Sensitive: true,
+			},
+
+			"domain": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"region": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Optional: true,
+				Default:  "us",
+			},
+		},
+	}
+}
+
+func resourceMailgunCredentialImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+
+	setDefaultRegionForImport(d)
+
+	log.Printf("[DEBUG] Import credential for region '%s' and email '%s'", d.Get("region"), d.Id())
+
+	return []*schema.ResourceData{d}, nil
+}
+
+func resourceMailgunCredentialCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, errc := meta.(*Config).GetClientForDomain(d.Get("region").(string), d.Get("domain").(string))
+	if errc != nil {
+		return diag.FromErr(errc)
+	}
+
+	email := fmt.Sprintf("%s@%s", d.Get("login").(string), d.Get("domain").(string))
+	password := d.Get("password").(string)
+
+	log.Printf("[DEBUG] Credential create configuration: {email: %s, password: %s}", email, password)
```

Comment:
Password is sensitive information. I don't know if it's good think to print this.